pages/wards/[id]/schedule-visit.js

```diff
@@ -258,36 +260,48 @@ const queryContainsInitialData = (query) => {
   return initialDataKeys.every((key) => queryKeys.includes(key));
 };
 
-export const getServerSideProps = verifyToken(
-  ({ query }) => {
-    const { id } = query;
-    let props = { id };
-    if (queryContainsInitialData(query)) {
-      const {
-        patientName,
-        contactName,
-        contactNumber,
-        day,
-        month,
-        year,
-        hour,
-        minute,
-      } = query;
-      const callDateTime = { day, month, year, hour, minute };
+export const getServerSideProps = propsWithContainer(
+  verifyToken(
+    async ({ query, container }) => {
+      const { id } = query;
+      let props = { id };
+      if (queryContainsInitialData(query)) {
+        const {
+          patientName,
+          contactName,
+          contactNumber,
+          day,
+          month,
+          year,
+          hour,
+          minute,
+        } = query;
+        const callDateTime = { day, month, year, hour, minute };
 
-      props = {
-        ...props,
-        initialPatientName: patientName,
-        initialContactName: contactName,
-        initialContactNumber: contactNumber,
-        initialCallDateTime: callDateTime,
-      };
+        props = {
+          ...props,
+          initialPatientName: patientName,
+          initialContactName: contactName,
+          initialContactNumber: contactNumber,
+          initialCallDateTime: callDateTime,
+        };
+      } else if (query.rebook) {
+        const { scheduledCall, error } = await retrieveVisitByCallId(container)(
+          query.rebook
```

Comment:
What do we think about renaming this to rebookCallId ? Otherwise the only way you know that this is actually a callId being passed through is by it being used in `retrieveVisitByCallId`

Reply:
Great idea - done :)